spec/models/concerns/date_accessible_spec.rb

```diff
@@ -1,162 +1,93 @@
-      expect(subject).to respond_to(:readwrite_multi_date_month=)
-      expect(subject).to respond_to(:readwrite_multi_date_year=)
+      expect(subject).to respond_to(:expiration_date_day=)
+      expect(subject).to respond_to(:expiration_date_month=)
+      expect(subject).to respond_to(:expiration_date_year=)
 
-      expect(subject).to respond_to(:readwrite_multi_second_date_day)
-      expect(subject).to respond_to(:readwrite_multi_second_date_month)
-      expect(subject).to respond_to(:readwrite_multi_second_date_year)
+      expect(subject).to respond_to(:issue_date_day)
+      expect(subject).to respond_to(:issue_date_month)
+      expect(subject).to respond_to(:issue_date_year)
 
-      expect(subject).to respond_to(:readwrite_multi_second_date_day=)
-      expect(subject).to respond_to(:readwrite_multi_second_date_month=)
-      expect(subject).to respond_to(:readwrite_multi_second_date_year=)
+      expect(subject).to respond_to(:issue_date_day=)
+      expect(subject).to respond_to(:issue_date_month=)
+      expect(subject).to respond_to(:issue_date_year=)
     end
   end
 
   describe "#date_writer" do
-    it 'should allow the use of the a _day writer' do
-      expect(subject).to respond_to(:write_date_day=)
-
-      expect(subject.write_date).to be_nil
-
-      subject.write_date_day = 12
-      expect(subject.write_date).to eq(Date.new.change(day: 12))
-    end
-
-    it 'should allow the use of the a _month writer' do
-      expect(subject).to respond_to(:write_date_month=)
-
-      expect(subject.write_date).to be_nil
-
-      subject.write_date_month = 5
-      expect(subject.write_date).to eq(Date.new.change(month: 5))
-    end
-
-    it 'should allow the use of the a _year writer' do
-      expect(subject).to respond_to(:write_date_year=)
-
-      expect(subject.write_date).to be_nil
-
-      subject.write_date_year = 1990
-      expect(subject.write_date).to eq(Date.new.change(year: 1990))
-    end
-
-    it 'should have garbage-resistant writers' do
-      subject.write_date_day = ''
-
-      expect(subject.write_date).to be_nil
-
-      subject.write_date_day = "foo"
-
-      expect(subject.write_date).to be_nil
-
-      subject.write_date_day = 0
-
-      expect(subject.write_date).to be_nil
-    end
-
-    it 'should not change the value when writing garbage' do
-      subject.write_date = Date.new(1990, 5, 12)
-
-      subject.write_date_day = "foo"
-
-      expect(subject.write_date).to eq(Date.new(1990, 5, 12))
+    it 'should allow the use of the a writer for all values' do
+      subject.expiration_date = nil
+      subject.expiration_date_day = 1
+      subject.expiration_date_month = 2
+      subject.expiration_date_year = year
+      subject.valid?
+      expect(subject.expiration_date).to eq(Date.new(year, 2, 1))
+
+      subject.expiration_date_day = 21
+      subject.expiration_date_year = nil
+      subject.valid?
+      expect(subject.expiration_date).to eq(Date.new(year, 2, 1))
```

Comment:
hmm i don't understand this example -- the year is `nil` but the date is still set to `year` -- is it b/c `year` `nil` is invalid so therefore it was never written to the expiration date?

Reply:
that's right, the intent is to say that b/c ``year` `nil` is invalid, it won't write to the date. I'll split this out for better clarity on what it's testing